src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -456,31 +457,56 @@ def nlcd_kfactor(result):
 
     return output
 
+def geop_tasks():
+    return geop_task_defs().keys()
+
+def geop_task_defs():
+    return {
+            'nlcd_streams': lambda geom: (
+                'nlcd_streams',
+                {'polygon': [geom.geojson], 'vector': streams(geom)},
+                nlcd_streams),
+            'nlcd_soils': lambda geom: (
+                'nlcd_soils',
+                {'polygon': [geom.geojson]},
+                nlcd_soils),
+            'gwn': lambda geom: ('gwn', {'polygon': [geom.geojson]}, gwn),
+            'avg_awc': lambda geom: (
+                'avg_awc',
+                {'polygon': [geom.geojson]},
+                avg_awc),
+            'nlcd_slope': lambda geom: (
+                'nlcd_slope',
+                {'polygon': [geom.geojson]},
+                nlcd_slope),
+            'slope': lambda geom: (
+                'slope',
+                {'polygon': [geom.geojson]},
+                slope),
+            'nlcd_kfactor': lambda geom: (
+                'nlcd_kfactor',
+                {'polygon': [geom.geojson]},
+                nlcd_kfactor)
+    }
 
-def geop_tasks(geom, errback, exchange, choose_worker):
-    # List of tuples of (opname, data, callback) for each geop task
-    definitions = [
-        ('nlcd_streams', {'polygon': [geom.geojson],
-                          'vector': streams(geom)}, nlcd_streams),
-        ('nlcd_soils', {'polygon': [geom.geojson]}, nlcd_soils),
-        ('gwn', {'polygon': [geom.geojson]}, gwn),
-        ('avg_awc', {'polygon': [geom.geojson]}, avg_awc),
-        ('nlcd_slope', {'polygon': [geom.geojson]}, nlcd_slope),
-        ('slope', {'polygon': [geom.geojson]}, slope),
-        ('nlcd_kfactor', {'polygon': [geom.geojson]}, nlcd_kfactor)
-    ]
-
-    tasks = zip(definitions, [choose_worker()
-                              for _ in xrange(len(definitions))])
-
-    return [(mapshed_start.s(opname, data).set(exchange=exchange,
-                                               routing_key=geop_worker) |
-             mapshed_finish.s().set(exchange=exchange,
-                                    routing_key=geop_worker) |
-             callback.s().set(link_error=errback,
-                              exchange=exchange,
-                              routing_key=choose_worker()))
-            for ((opname, data, callback), geop_worker) in tasks]
+@shared_task(bind=True, default_retry_delay=1, max_retries=42)
```

Comment:
I don't think we should be retrying the entire chain of `geop_task`. The `mapshed_start` and `mapshed_finish` tasks have retries specified on them and that should be enough.